internal/pkg/manifest/workload_applyenv_test.go

```diff
@@ -0,0 +1,1846 @@
+			inSvc: func(svc *LoadBalancedWebService) {
+				svc.ImageConfig.Build = BuildArgsOrString{
+					BuildArgs: DockerBuildArgs{
+						Target: aws.String("mockTarget"),
+					},
+				}
+				svc.Environments["test"].ImageConfig.Build = BuildArgsOrString{
+					BuildArgs: DockerBuildArgs{
+						Target: aws.String("mockTargetTest"),
+					},
+				}
+			},
+			wanted: func(svc *LoadBalancedWebService) {
+				svc.ImageConfig.Build = BuildArgsOrString{
+					BuildString: nil,
+					BuildArgs: DockerBuildArgs{
+						Target: aws.String("mockTargetTest"),
+					},
+				}
+			},
+		},
+		"target explicitly overridden by empty value": {
+			inSvc: func(svc *LoadBalancedWebService) {
+				svc.ImageConfig.Build = BuildArgsOrString{
+					BuildArgs: DockerBuildArgs{
+						Target: aws.String("mockTarget"),
+					},
+				}
+				svc.Environments["test"].ImageConfig.Build = BuildArgsOrString{
+					BuildArgs: DockerBuildArgs{
+						Target: aws.String(""),
+					},
+				}
+			},
+			wanted: func(svc *LoadBalancedWebService) {
+				svc.ImageConfig.Build = BuildArgsOrString{
+					BuildString: nil,
+					BuildArgs: DockerBuildArgs{
+						Target: aws.String(""),
+					},
+				}
+			},
+		},
+		"target not overridden": {
+			inSvc: func(svc *LoadBalancedWebService) {
+				svc.ImageConfig.Build = BuildArgsOrString{
+					BuildArgs: DockerBuildArgs{
+						Target: aws.String("mockTarget"),
+					},
+				}
+				svc.Environments["test"].ImageConfig.Build = BuildArgsOrString{
+					BuildArgs: DockerBuildArgs{},
+				}
+			},
+			wanted: func(svc *LoadBalancedWebService) {
+				svc.ImageConfig.Build = BuildArgsOrString{
+					BuildString: nil,
+					BuildArgs: DockerBuildArgs{
+						Target: aws.String("mockTarget"),
+					},
+				}
+			},
+		},
+		"cacheFrom overridden": {
+			inSvc: func(svc *LoadBalancedWebService) {
+				svc.ImageConfig.Build = BuildArgsOrString{
+					BuildArgs: DockerBuildArgs{
+						CacheFrom: []string{"mock", "Cache"},
+					},
+				}
+				svc.Environments["test"].ImageConfig.Build = BuildArgsOrString{
+					BuildArgs: DockerBuildArgs{
+						CacheFrom: []string{"mock", "CacheTest", "Test"},
+					},
+				}
+			},
+			wanted: func(svc *LoadBalancedWebService) {
+				svc.ImageConfig.Build = BuildArgsOrString{
+					BuildString: nil,
+					BuildArgs: DockerBuildArgs{
+						CacheFrom: []string{"mock", "CacheTest", "Test"},
+					},
+				}
+			},
+		},
+		"cacheFrom explicitly overridden by empty value": {
+			inSvc: func(svc *LoadBalancedWebService) {
+				svc.ImageConfig.Build = BuildArgsOrString{
+					BuildArgs: DockerBuildArgs{
+						CacheFrom: []string{"mock", "Cache"},
+					},
+				}
+				svc.Environments["test"].ImageConfig.Build = BuildArgsOrString{
+					BuildArgs: DockerBuildArgs{
+						CacheFrom: []string{},
+					},
+				}
+			},
+			wanted: func(svc *LoadBalancedWebService) {
+				svc.ImageConfig.Build = BuildArgsOrString{
+					BuildString: nil,
+					BuildArgs: DockerBuildArgs{
+						CacheFrom: []string{},
```

Comment:
Is this the behavior that we want for `slices`? is it consistent everywhere today?  
I'm just asking because it's not what Compose does :( where they strictly just append to the existing slice.

Reply:
Good call! All slice override should be "append"

Reply:
Addressed!